### PR TITLE
Initial frontend angular

### PR DESCRIPTION
**Pull request by brtsai**
Most of the front-end has been setup using angular4 for everything up until the user has logged in.
Do not currently know how to detect if the user is logged in or not.

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+<app-navbar> </app-navbar>
+
+<router-outlet> </router-outlet>
```
Comment (altech-stack) [inline src/app/app.component.html:4]: Good organized code here.

File: src/app/app.module.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    SignupComponent,
+    HomepageComponent,
+    FooterComponent,
+    ContactComponent
+  ],
+  imports: [
+    BrowserModule,
+    FormsModule,
+    HttpModule,
+    RouterModule.forRoot([
```
Comment (altech-stack) [inline src/app/app.module.ts:29]: Nice use of the router!

File: src/app/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+
+  constructor() { }
+
```
Comment (altech-stack) [inline src/app/login/login.component.ts:11]: Cool, so this is where we'll put in the logic for the login request, which I'll tie up in the backend

**Comment by altech-stack** [comment]
Nice code changes. I should be able to work on backend today or monday!

File: src/app/app.component.html (hunk 1 above)
Comment (brtsai) [reply src/app/app.component.html:4]: Thanks! I remembered that on Saturday you mentioned the main app component should only import other components.

File: src/app/app.module.ts (hunk 2 above)
Comment (brtsai) [reply src/app/app.module.ts:29]: Thanks, the instructional video you linked me really helped!

File: src/app/login/login.component.ts (hunk 3 above)
Comment (brtsai) [reply src/app/login/login.component.ts:11]: Ah. How do I make calls to the API/backend from Angular?

**Pull request merged**